Using xor calculating hamming distance is blazingly fast

### Xor has following properties:

For A xor B = C, have A xor C = B, and B xor A = C.

***

In python3, A xor B is represent as follows where both A,B are in decimal base:
       
       A^B
       
For example,

In [1]:
666^777

403

666 -> 1010011010

-777 ->-1100001001

====0110010011 #same is 0, differ is 1

0110010011->403 # hamming_dist = 5

### Binary manipulation

In [18]:
x = int('1010011010  ', 2)
y = int('1100001001  ', 2)
print(x)
print(y)
z = x^y
print(z)
print(bin(z)[2:])
print(type(bin(z)[2:]))
print(bin(z)[2:].count('1'))
print('1100001001'.count('1'))



666
777
403
110010011
<class 'str'>
5
4
